### DISCUSSION TITLE
unfinished hook action

### DISCUSSION THREAD

**Discussion by bitdom8** [sveltejs/kit#5283] [General]
After updating 352 to 353, received an error when run build:

[!] Error: unfinished hook action(s) on exit:
(vite-plugin-svelte-kit) writeBundle

**Comment by benmccann** [comment]
There's not enough info here to diagnose your issue. You'd need to provide a repository with the project's code

**Reply by tsukhu** [reply]
I am also facing the same issue and build error with the latest version of sveltekit. This comes only during the production builds and my vercel deployment is also failing with the same issue. I have reverted back to version 350 and then the build and deployment succeeeds

**Comment by bitdom8** [comment]
True, you can also change back to 352, all is fine. I have Win10 and my package and svelte.config as follows:

```
{
  "name": "clienttry",
  "version": "1.0.3",
  "private": true,
  "scripts": {
    "dev": "svelte-kit dev",
    "build": "svelte-kit build --verbose",
    "package": "svelte-kit package",
    "previewtrylinux": "svelte-kit preview --host=91.226.221.182 --port=2999",
    "previewtry": "svelte-kit preview",
    "start": "node ./server.js",
    "preview": "svelte-kit preview --host=91.226.221.182 --port=3000",
    "preview2": "svelte-kit preview --host=91.226.221.182 --port=3001",
    "prepare": "svelte-kit sync",
    "check": "svelte-check --tsconfig ./tsconfig.json",
    "check:watch": "svelte-check --tsconfig ./tsconfig.json --watch",
    "lint": "prettier --ignore-path .gitignore --check --plugin-search-dir=. . && eslint --ignore-path .gitignore .",
    "format": "prettier --ignore-path .gitignore --write --plugin-search-dir=. .",
    "typesafe-i18n": "typesafe-i18n"
  },
  "devDependencies": {
    "@sveltejs/adapter-auto": "next",
    "@sveltejs/adapter-node": "1.0.0-next.78",
    "@sveltejs/kit": "1.0.0-next.352",
    "@types/cookie": "^0.5.1",
    "@typescript-eslint/eslint-plugin": "5.30.0",
    "@typescript-eslint/parser": "5.30.0",
    "eslint": "8.18.0",
    "eslint-config-prettier": "^8.5.0",
    "eslint-plugin-svelte3": "^4.0.0",
    "prettier": "2.7.1",
    "prettier-plugin-svelte": "2.7.0",
    "svelte": "3.48.0",
    "svelte-check": "^2.8.0",
    "svelte-preprocess": "4.10.7",
    "tslib": "^2.4.0",
    "typescript": "4.7.4"
  },
  "type": "module",
  "dependencies": {
    "@zerodevx/svelte-toast": "^0.7.2",
    "body-parser": "^1.20.0",
    "cookie": "^0.5.0",
    "dotenv": "^16.0.1",
    "express": "^4.18.1",
    "filepond": "^4.30.4",
    "filepond-plugin-image-exif-orientation": "^1.0.11",
    "filepond-plugin-image-preview": "4.6.11",
    "filepond-plugin-image-resize": "^2.0.10",
    "filepond-plugin-image-transform": "^3.8.7",
    "fuse.js": "^6.6.2",
    "marked": "^4.0.17",
    "mongodb": "^4.7.0",
    "nanoid": "^4.0.0",
    "node-cron": "^3.0.1",
    "node-telegram-bot-api": "^0.58.0",
    "pnpm": "^7.3.0",
    "puppeteer": "^15.1.1",
    "push.js": "^1.0.12",
    "sass": "1.53.0",
    "svelte-intersection-observer": "^0.10.0",
    "twitter-api-v2": "^1.12.2",
    "typesafe-i18n": "^5.5.2",
    "ws": "^8.8.0"
  }
}

```

```
import adapter from '@sveltejs/adapter-node';
import preprocess from 'svelte-preprocess';
import path from 'path'

export default {
	kit: {
		  adapter: adapter({ }),
		  alias: {
			$i18n: path.resolve('./src/i18n'),
			$src: path.resolve('./src'),
		},
	},
		preprocess : preprocess({
		preserve: ['ld+json'],
	  }),
  };

```

**Comment by bitdom8** [comment]
@benmccann Thanks

**Comment by tsukhu** [comment]
Any update on this issue. I haven't tried out the latest version to check thought

**Reply by tsukhu** [reply]
@benmccann - Did some more investigation. See if this is of any help to find the issue

Here is where the build is failing. It seems to complete everything and then fails
![image](https://user-images.githubusercontent.com/846306/176601327-1662fd73-ce03-4bfb-b827-05db19456d02.png)

Looking at the code
https://github.com/sveltejs/kit/blob/master/packages/kit/src/vite/index.js#L305-L311

I couldn't find any adapt folder in my build. Unless I am missing something.
```javascript
if (svelte_config.kit.adapter) {
				const { adapt } = await import('./build/adapt/index.js');
				await adapt(svelte_config, build_data, prerendered, { log });

				// this is necessary to close any open db connections, etc
				process.exit(0);
			}
```

**Reply by bitdom8** [reply]
It works well with the real svelteworld but not with my app any more https://github.com/sveltejs/realworld

**Reply by tsukhu** [reply]
Hmm I guess what I provided yesterday is just the internal reference of the sveltekit reference and does clear that part as I do get the done text .

Will see if I find anything else. During the vite build is there a way to have additional debug information displayed, to help figure out this issue

**Reply by tsukhu** [reply]
Hi @bitdom8  @benmccann ,

Taking a clue from the reference to the realworld app. I deleted the node_modules , removed the yarn lock file and created it. Now it works with the latest version . Looks like the dependencies in the lock file were not in sync . Now it builds both with the `node` and `vercel` adapter. Mine is a very old project which I keep upgrading .

**Comment by bitdom8** [comment]
Hi @tsukhu @benmccann ,

using pnpm deleted pnpm lock and modules. And reinstall npm instead. Works now. You sir are remarkable.
